lib/nokogiri/xml/node.rb

```diff
@@ -191,17 +197,70 @@ def decorate!
       end
 
       # :section: Manipulating Document Structure
-
-      ###
-      # Add +node_or_tags+ as a child of this Node.
+      0
+      # :call-seq:
+      #   add_child(object) -> Node or NodeSet
       #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
+      # Appends specified Nodes to the children of +self+;
+      # each appended Node has +self+ as its #parent value,
+      # and <tt>self.document</tt> as its #document value.
       #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
+      # [Arguments]
+      #
+      # - +object+ (Node, NodeSet, DocumentFragment, String):
+      #   specifies the Node objects to be appended;
+      #   specified Nodes may be in the same Document or DocumentFragment as +self+,
+      #   or in a different one.
+      #
+      # [Returns]
+      #
+      # - The given +object+, if +object+ is a Node or a NodeSet
+      # - A new NodeSet, if +object+ is a DocumentFragment or String.
+      #
+      # When +object+ is a +Node+,
+      # adds it as the last child of +self+;
+      # returns +object+:
+      #
+      #   xml = '<root><src_parent><src_child/></src_parent><dst_parent><dst_child/></dst_paren></root>'
+      #   doc = Nokogiri::XML::Document.parse(xml)
+      #   src_child_node = doc.at_css('src_child')
+      #   dst_parent_node = doc.at_css('dst_parent')
+      #   dst_parent_node.add_child(src_child_node)
+      #
+      # When +object+ is a NodeSet (see {About the Examples}[Node.html#class-Nokogiri::XML::Node-label-About+the+Examples]),
+      # appends each of its nodes to the children of +self+;
+      # returns +object+:
+      #
+      #   bookstore_doc = Nokogiri::XML::Document.parse(BOOKSTORE_XML)
+      #   book_nodeset = bookstore_doc.search('//book')
```

Comment:
I'd prefer if we did not use `search` in examples. I've been trying to discourage the use of that method for many years. Let's please use either `xpath` or `css`.

Reply:
@flavorjones, going forward, do have a preference between the two?

Reply:
I don't have a preference, honestly. Depending on the task, an xpath query or a css selector might be easier ... Or if we want to cater to the reader, XML examples might use XPath, and HTML examples might use CSS?

Reply:
Interesting, thanks.

Reply:
Fixed.